### DISCUSSION TITLE
Value Error: Unable to expand environment variable

### DISCUSSION THREAD

**Discussion by oceanKingEth** [smartcontractkit/full-blockchain-solidity-course-py#20] [Q&A] [answered]
Hello,
I am having difficulty connecting to the Rinkeby network through infura and brownie.
This happened at the end of lesson 4 and now lesson 5.
I am able to deploy and test on ganache, but never rinkeby.
I saw this problem come up on stack exchange and perhaps I need to upgrade versions of brownie? i have Brownie v1.14.4

I am confident I have the .env and brownie-config.yaml  files set up correctly.

Thanks for helping

BrownieFundMeProject is the active project.
  File "brownie/_cli/__main__.py", line 64, in main
    importlib.import_module(f"brownie._cli.{cmd}").main()
  File "brownie/_cli/run.py", line 43, in main
    network.connect(CONFIG.argv["network"])
  File "brownie/network/main.py", line 40, in connect
    web3.connect(host, active.get("timeout", 30))
  File "brownie/network/web3.py", line 51, in connect
    uri = _expand_environment_vars(uri)
  File "brownie/network/web3.py", line 171, in _expand_environment_vars
    raise ValueError(f"Unable to expand environment variable in host setting: '{uri}'")
ValueError: Unable to expand environment variable in host setting: 'https://rinkeby.infura.io/v3/$WEB3_INFURA_PROJECT_ID'

**Comment by oceanKingEth** [comment, marked as answer]
I figured it out!!!!! Man, that took me hours. lol
i ended up using source .env in the terminal, seems to work now.

EDIT: This won't work for Windows, and means you'll have to run this everytime you start a project. Ideally you add `dotenv: .env` to your `brownie-config.yaml` and add your environment variables to a `.env` file.

**Reply by PatrickAlphaC** [reply]
Thank you!!

What timestamp is this at? You should be able to add `dotenv: .env` to your `brownie-config.yaml` and it should have the same affect.

I LOVE seeing you go through this by the way!

**Reply by PatrickAlphaC** [reply]
That `source .env` trick will mean you have to do it everytime you close your terminal, and doesn't work for windows. So I'm wondering if I forgot to tell people to add `.env` to their brownie config?

**Reply by oceanKingEth** [reply]
you definetely showed us to add dotenv: .env to the config file, i'm not sure what happened there.
thanks for responding and for the course itself! i'm loving this already.

**Reply by cherb-programs** [reply]
Thanks guys, worked a treat.

**Reply by haraslub** [reply]
I have just faced to the same issue. Thank both of you @PatrickAlphaC and @oceanKingEth for the solution provided.

**Reply by cabelera42** [reply]
the timestamp is 7:54:13

the error first shows when running: brownie test -k test_get_entrance_fee --network rinkeby
